packages/core/src/Tooltip/index.tsx

```diff
@@ -194,11 +197,21 @@ export const Tooltip: React.FC<TooltipProps | ExpandedTooltipProps> = ({
     const tooltipEl = tooltipRef.current
 
     const { bottom } = anchorEl.getBoundingClientRect()
+    const { left, right } = tooltipEl.getBoundingClientRect()
+
     const bottomSpace = document.documentElement.clientHeight - bottom
     // See if `bottomSpace` is smaller than Tooltip height.
-    // "8" is margin of the TooltipWapper.
+    // "8" is margin of the TooltipWrapper.
     setHasOverflow(tooltipEl.clientHeight + 8 > bottomSpace)
-  }, [anchorEl, tooltipRef, hide, show])
+
+    if (horizontalLayout === 'center') {
+      if (left < 0) {
+        setHorizontalLayout('left')
+      } else if (right > document.documentElement.clientWidth) {
+        setHorizontalLayout('right')
+      }
+    }
+  }, [anchorEl, tooltipRef, hide, show, horizontalLayout])
```

Comment:
We had `hide` and `show` in the deps array, but they were not used?

Reply:
Removed.